src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -76,7 +76,12 @@
   `(loop [pool-instance# (jruby/borrow-instance ~jruby-service)]
      (if (nil? pool-instance#)
        (throw (IllegalStateException.
-                "Error: Attempt to borrow a JRuby instance from the pool timed out")))
```

Comment:
Since this is now an `exception` that we'd be throwing and recatching from within Puppet Server, might be good to turn this into a Slingshot exception - like we do with the "bad requests" in the request-handler-core namespace.